deploy/kubernetes/operator/pkg/webhook/inspector/rss.go

```diff
@@ -35,19 +35,19 @@ import (
 // validateRSS validates the create and update operation towards rss objects.
 func (i *inspector) validateRSS(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionReview {
 	if !i.ignoreRSS && ar.Request.Operation == admissionv1.Update {
-		oldRSS := &unifflev1alpha1.RemoteShuffleService{}
+		oldRSS := &uniffleAPI.RemoteShuffleService{}
 		if err := json.Unmarshal(ar.Request.OldObject.Raw, oldRSS); err != nil {
 			klog.Errorf("unmarshal old object of rss (%v) failed: %v",
 				string(ar.Request.OldObject.Raw), err)
 			return util.AdmissionReviewFailed(ar, err)
 		}
-		// for security purposes, we forbid updating rss objects when they are in upgrading phase.
-		if oldRSS.Status.Phase == unifflev1alpha1.RSSUpgrading {
-			message := "can not update upgrading rss object: " + utils.UniqueName(oldRSS)
+		// for security purposes, we forbid updating rss objects when they are in upgrading/terminating phase.
+		if oldRSS.Status.Phase == uniffleAPI.RSSUpgrading || oldRSS.Status.Phase == uniffleAPI.RSSTerminating {
```

Comment:
I think the following should be allowed:
oldRSS.Status.Phase == uniffleAPI.RSSUpgrading && newRSS.status == uniffleAPI.RSSTerminating

Reply:
> Let me confirm one thing first, when controller updates rss' upgrading to terminating, is that update request also validated by this code path?
> 
> In my local test, this line was kept intact, I only changed controller/rss.go:L357, and found upgrading rss object could be deleted normally.

Sorry, webhook is not currently intercepting the subresource status, so updatestatus is not affected. But after Terminating, there will be an action to delete finalizer. Is this also not blocked in your test environment ?

Reply:
In a very early version, validatingwebhookconfiguration object contains "remoteshuffleservices/status" (deploy/kubernetes/operator/PKG/webhook/syncer/syncer.go: 244), I have forgotten it.

Reply:
The only protection mechanism is finalizer. This is not automatically added by the k8s, but by the controller. Controller needs to add and remove finalizers.